### PR TITLE
Implement storing of IP parameters between runs

### LINKED ISSUE #92: StationName and IP address are not stored between runs
The OSAL API needs to be updated

This is indicated by for example Automated RT Tester DCP_IP_parameters_remanence

### PR DESCRIPTION
**Pull request by pyhys**
OSAL functions to save and load binary blobs

Add function to get MAC address from config

Improve logging of IP addresses

Rename variables to better reflect their usage

Change revision prefix in sample application

Closes #92

### CODE REVIEW

File: sample_app/sampleapp_common.c [hunk 1]
```diff
@@ -365,6 +365,10 @@ static int app_reset_ind(
          reset_mode);
    }
 
+   if (should_reset_application == true){
+         os_clear_blob(PNET_FILE_INDEX_USER);
```
Comment (Tjohej) [inline sample_app/sampleapp_common.c:369]: Cant we call it os_clear_file? or os_clear_info...

File: src/device/pf_cmina.c [hunk 2]
```diff
@@ -21,24 +21,29 @@
...
 #include "pf_includes.h"
 
-static const char             *hello_sync_name = "hello";
+static const char          *hello_sync_name = "hello";
 
 
 /**
  * @internal
- * Send a DCP HELLO message.
+ * Trigger sending a DCP HELLO message.
```
Comment (Tjohej) [inline src/device/pf_cmina.c:46]: This new text suggests (to me) that it simply sets a flag and then "somebody else" does the sending, but pf_dcp_hello_req actually sends the request.
I think we should keep the previous text.

File: src/device/pf_cmina.c [hunk 3]
```diff
@@ -73,6 +78,69 @@ static void pf_cmina_send_hello(
...
+ * Compares with the content of already stored settings (in order not to
+ * wear out the flash chip)
+ *
+ * @param p_ase            In:   Settings to be saved
+ */
+static void pf_cmina_save_ase_if_modified(
+   pf_cmina_dcp_ase_t     *p_ase)
+{
+   pf_cmina_dcp_ase_t      file_ase;
+   bool                    should_save = false;
```
Comment (olbjo) [inline src/device/pf_cmina.c:94]: should_save -> save

File: src/device/pf_cmina.c [hunk 4]
```diff
@@ -73,6 +78,69 @@ static void pf_cmina_send_hello(
...
+   }
+   else
+   {
+      LOG_DEBUG(PF_DCP_LOG,"CMINA(%d): No storing of nvm IP settings (no changes). "
+         "IP: %s Netmask: %s Gateway: %s Station name: %s\n", __LINE__,
+         ip_string, netmask_string, gateway_string, p_ase->name_of_station);
+   }
+}
+
 int pf_cmina_set_default_cfg(
```
Comment (olbjo) [inline src/device/pf_cmina.c:144]: Not so important, but maybe function name is misleading. Function is resetting config parameters according to reset mode parameter. It is not only applying a default configuration.

File: src/osal/linux/osal.c [hunk 5]
```diff
@@ -35,13 +35,118 @@
...
+   const char              *s = "unknown";
+
+   switch (file_index)
+   {
+   case 0:  s = "pnet_data_ip.bin"; break;
+   case 1:  s = "pnet_data_diagnostics.bin"; break;
+   case 2:  s = "pnet_data_logbook.bin"; break;
+   case 3:  s = "pnet_data_user.bin"; break;
+   default:
+      break;
```
Comment (olbjo) [inline src/osal/linux/osal.c:61]: why new line for this break?

File: src/osal.h [hunk 6]
```diff
@@ -114,6 +116,45 @@ enum os_eth_type {
...
+   void                    *object,
+   size_t                  size
+);
+
+/**
+ * Clear a binary blob.
+ *
+ * @param file_index       In: Index of binary blobs
+ */
+void os_clear_blob(
```
Comment (olbjo) [inline src/osal.h:154]: Maybe os_delete_blob()? Clear might indicate that a cleared blob can be read and a "cleared" blob is read back

File: src/device/pf_cmina.c (hunk 2 above)
Comment (pyhys) [reply src/device/pf_cmina.c:46]: Yes, this text is for pf_cmina_send_hello().

File: src/osal.h [hunk 7]
```diff
@@ -114,6 +116,45 @@ enum os_eth_type {
...
 typedef uint16_t os_ipport_t;
 
+/**
+ * Load a binary blob.
+ *
+ * @param file_index       In:    Index of binary blobs
+ * @param object           Out:   Struct to load
+ * @param size             In:    Size of struct to load
+ * @return  0  if the operation succeeded.
+ *          -1 if an error occurred.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/osal.h:126]: Done

File: src/device/pf_cmina.h [hunk 8]
```diff
@@ -62,6 +71,17 @@ int pf_cmina_get_ipaddr(
    pnet_t                  *net,
    os_ipaddr_t             *p_ipaddr);
 
+/**
+ * Retrieve the MAC address.
+ * @param net              InOut: The p-net stack instance
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.h:76]: Done

File: sample_app/sampleapp_common.c (hunk 1 above)
Comment (pyhys) [reply sample_app/sampleapp_common.c:369]: Removed the user part of the storage

File: src/device/pf_cmina.h (hunk 8 above)
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.h:76]: Done

File: src/osal.h [hunk 9]
```diff
@@ -114,6 +116,45 @@ enum os_eth_type {
 typedef uint32_t os_ipaddr_t;
 typedef uint16_t os_ipport_t;
 
+/**
+ * Load a binary blob.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/osal.h:120]: Added a todo

File: src/device/pf_cmina.c (hunk 3 above)
Comment (pyhys) [reply src/device/pf_cmina.c:94]: Done

File: src/osal/linux/osal.c (hunk 5 above)
Comment (pyhys) [reply src/osal/linux/osal.c:61]: Done

**Comment by pyhys** [comment]
Suggested new name:
os_save_file( const char *filname, const void *pData, size_t dataSize)

**Pull request merged**